src/Extensions/Compatability/Validation/Code.cs

```diff
@@ -0,0 +1,510 @@
+		/// <typeparam name="T">The element type.</typeparam>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static bool ValidateAny<T>([NotNullWhen(true)]IEnumerable<T>? argumentValue, string argumentName, EventId? tagId = null)
+		{
+			if (!ValidateArgument(argumentValue, argumentName, tagId))
+			{
+				return false;
+			}
+
+			if (!argumentValue.Any())
+			{
+				LogError(tagId, ValidationFailed, HasAnyErrorMessage, argumentName);
+				return false;
+			}
+
+			return true;
+		}
+
+
+		/// <summary>
+		/// Validates the argument isn't null.
+		/// </summary>
+		/// <param name="argumentValue">The argument value.</param>
+		/// <param name="argumentName">Name of the argument.</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		/// <returns>True if the argument <paramref name="argumentValue"/> is not null; false otherwise.</returns>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static bool ValidateArgument([NotNullWhen(true)]object? argumentValue, string argumentName, EventId? tagId = null)
+		{
+			if (argumentValue == null)
+			{
+				LogError(tagId, ValidationFailed, ArgumentIsNull, argumentName);
+				return false;
+			}
+
+			return true;
+		}
+
+
+		/// <summary>
+		/// Validates argument against a predicate
+		/// </summary>
+		/// <param name="argumentValue">argument value</param>
+		/// <param name="argumentName">argument name</param>
+		/// <param name="predicate">predicate to evaluate</param>
+		/// <param name="errorMessage">Error message for failed assertion</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		/// <typeparam name="T">Type of argument to validate</typeparam>
+		/// <returns>True if the argument <paramref name="argumentValue"/> passes predicate <paramref name="predicate"/>; false otherwise.</returns>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static bool ValidateArgument<T>(T argumentValue, string argumentName, Func<T, bool> predicate, string errorMessage, EventId? tagId = null)
+		{
+			if (!predicate(argumentValue))
+			{
+				LogError(tagId, ValidationFailed, errorMessage, argumentName);
+				return false;
+			}
+
+			return true;
+		}
+
+
+		/// <summary>
+		/// Validates the guid argument and returns false if it is empty.
+		/// </summary>
+		/// <param name="argumentValue">The argument value.</param>
+		/// <param name="argumentName">The argument name.</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		/// <returns>True if the argument <paramref name="argumentValue"/> is not an empty guid; false otherwise.</returns>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static bool ValidateArgument(Guid argumentValue, string argumentName, EventId? tagId = null)
+		{
+			if (argumentValue.Equals(Guid.Empty))
+			{
+				LogError(tagId, ValidationFailed, ArgumentIsEmptyGuid, argumentName);
```

Comment:
I would like to avoid adding new functionality here since the idea is to deprecate this approach.
We should think about another approach to validate arguments, maybe simular methods that accept logger or common methods for throwing validation exceptions.